### ISSUE TITLE
Unable to attach volume in block mode

### ISSUE THREAD

**Issue by rkrzewski** [topolvm/topolvm#406]
**Describe the bug**
`topolvm-node` reports that `NodePublishVolume(block)` command succeeded, but no device node is created in `/var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/<pv>`

**Environments**
- Version: topolvm 0.10.2, microk8s v1.21.7
- OS: Ubuntu 20.04.3 LTS, Linux x86_64 5.4.0-91-generic

**To Reproduce**
While trying to start a Ceph OSD pod, that is using topolvm volume in block mode I see the following events:
```
  Type     Reason           Age                   From     Message
  ----     ------           ----                  ----     -------
  Warning  FailedMount      42m (x3 over 76m)     kubelet  Unable to attach or mount volumes: unmounted volumes=[set1-data-1-7b67p], unattached volumes=[rook-ceph-log rook-ceph-crash set1-data-1-7b67p run-udev set1-data-1-7b67p-bridge kube-api-access-gf9fb rook-data rook-config-override]: timed out waiting for the condition
  Warning  FailedMapVolume  109s (x50 over 87m)   kubelet  MapVolume.EvalHostSymlinks failed for volume "pvc-a0365ed0-9ff9-4b5f-947d-9ec87f6dcee8" : lstat /var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/pvc-a0365ed0-9ff9-4b5f-947d-9ec87f6dcee8/0485c803-5831-43d6-b8e6-4cc40363d2c6: no such file or directory
```
Directory `/var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/pvc-a0365ed0-9ff9-4b5f-947d-9ec87f6dcee8/` exists but `0485c803-5831-43d6-b8e6-4cc40363d2c6` entry is in fact missing.

When looking for messages mentioning PV `pvc-a0365ed0-9ff9-4b5f-947d-9ec87f6dcee8` in the logs of `topolvm-node` container, I see the following:
```
{"level":"info","ts":1638540728.8841894,"logger":"driver.node","msg":"NodePublishVolume called","volume_id":"7bf6681e-8015-4c37-9b6f-b22cd0c6de9a","publish_context":null,"target_path":"/var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/pvc-a0365ed0-9ff9-4b5f-947d-9ec87f6dcee8/0485c803-5831-43d6-b8e6-4cc40363d2c6","volume_capability":"block:{}  access_mode:{mode:SINGLE_NODE_WRITER}","read_only":false,"num_secrets":0,"volume_context":{"storage.kubernetes.io/csiProvisionerIdentity":"1609003784254-8081-topolvm.cybozu.com"}}
{"level":"info","ts":1638540729.056411,"logger":"driver.node","msg":"NodePublishVolume(block) succeeded","volume_id":"7bf6681e-8015-4c37-9b6f-b22cd0c6de9a","target_path":"/var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/pvc-a0365ed0-9ff9-4b5f-947d-9ec87f6dcee8/0485c803-5831-43d6-b8e6-4cc40363d2c6"}
```
Followed by more `NodePublishVolume called` entries with the same parametrs, but no more  `NodePublishVolume(block) succeeded` for the same PV.

I have another cluster running rook on top of topolvm 0.6.2 and I can see that `topolvm-node` creates block device entries under `csi/volumeDevices/publish/<pv>`.

Also, on the cluster running topolvm 0.10.2, a topolvm volume that is being used by Ceph Mon pod in filesystem mode is published correctly. Here are relevant entries from `topolvm-node` log:
```
{"level":"info","ts":1638391065.2716677,"logger":"driver.node","msg":"NodePublishVolume called","volume_id":"c9f94989-6a8e-4ba0-9d45-67648090b4bf","publish_context":null,"target_path":"/var/snap/microk8s/common/var/lib/kubelet/pods/0cbfa428-f237-47a6-b30a-bcfbba4e619d/volumes/kubernetes.io~csi/pvc-46fbd9a6-b0e3-46e3-9e18-60f36fe95cb7/mount","volume_capability":"mount:{fs_type:\"ext4\"}  access_mode:{mode:SINGLE_NODE_WRITER}","read_only":false,"num_secrets":0,"volume_context":{"csi.storage.k8s.io/ephemeral":"false","csi.storage.k8s.io/pod.name":"rook-ceph-mon-a-754d5996bd-lln2n","csi.storage.k8s.io/pod.namespace":"rook-ceph","csi.storage.k8s.io/pod.uid":"0cbfa428-f237-47a6-b30a-bcfbba4e619d","csi.storage.k8s.io/serviceAccount.name":"default","storage.kubernetes.io/csiProvisionerIdentity":"1609003784254-8081-topolvm.cybozu.com"}}
{"level":"info","ts":1638391065.426326,"logger":"driver.node","msg":"NodePublishVolume(fs) succeeded","volume_id":"c9f94989-6a8e-4ba0-9d45-67648090b4bf","target_path":"/var/snap/microk8s/common/var/lib/kubelet/pods/0cbfa428-f237-47a6-b30a-bcfbba4e619d/volumes/kubernetes.io~csi/pvc-46fbd9a6-b0e3-46e3-9e18-60f36fe95cb7/mount","fstype":"ext4"}
```
The directory exits and Mon pod starts up correctly.

**Expected behavior**
I would expect the block device node to be created under `csi/volumeDevices/publish` and successfully mounted in the consumer pod.

**Additional context**
This is a single-node experimental cluster that I've been trying to upgrade to microk8s 1.22. I found out that topolvm 0.6.2 is not compatible with k8s 1.22 because it's using a deprecated API that was removed in 1.22 (I think it was CustomResourceDefinition apiextensions.k8s.io/v1beta1).
I've tried to upgrade topolvm to 10.0.2 on that cluster. Since the deployment is now Helm based, I have modified many resources manually to be "adopted" by Helm by adding labels and annotations and then removed some old resources under different names. Another thing I needed to tweak was CSI-related paths. I'm using Ubuntu's microk8s and `/var/lib/kubelet` is actually `/var/snap/microk8s/common/var/lib/kubelet/` on my machines. The upgrade seems to have _mostly_ worked - all topolvm pod are starting up and aren't logging errors. Never the less the described problem tripped me up and I need to get it resolved before I can get to upgrading my other 3 node cluster. In conclusion: I am not sure if messed something up with my upgrade attempt or I if I uncovered a legitimate bug and I'll appreciate any pointers!

**Comment by rkrzewski** [comment]
I was able to track down and fix the issue in my cluster. This is caused by the `/var/lib/kubelet/` / `/var/snap/microk8s/common/var/lib/kubelet/` path mixup. The command `NodePublishVolume` contains a path from the perspective of the node, and the `csi-plugin-dir` volume must be mounted accordingly.

I had to configure the following volumes and volumeMounts for the Helm chart:
```
node:
  kubeletRegistrationPath: /var/snap/microk8s/common/var/lib/kubelet/plugins/topolvm.cybozu.com/node/csi-topolvm.sock
  volumes:
    - name: registration-dir
      hostPath:
        path: /var/snap/microk8s/common/var/lib/kubelet/plugins_registry/
        type: Directory
    - name: node-plugin-dir
      hostPath:
        path: /var/snap/microk8s/common/var/lib/kubelet/plugins/topolvm.cybozu.com/node
        type: DirectoryOrCreate
    - name: csi-plugin-dir
      hostPath:
        path: /var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi
        type: DirectoryOrCreate
    - name: pod-volumes-dir
      hostPath:
        path: /var/snap/microk8s/common/var/lib/kubelet/pods/
        type: DirectoryOrCreate
    - name: lvmd-socket-dir
      hostPath:
        path: /run/topolvm
        type: Directory
  volumeMounts:
    topolvmNode:
      - name: node-plugin-dir
        mountPath: /run/topolvm
      - name: lvmd-socket-dir
        mountPath: /run/lvmd
      - name: pod-volumes-dir
        mountPath: /var/snap/microk8s/common/var/lib/kubelet/pods
        mountPropagation: "Bidirectional"
      - name: csi-plugin-dir
        mountPath: /var/snap/microk8s/common/var/lib/kubelet/plugins/kubernetes.io/csi
        mountPropagation: "Bidirectional"
```
Moreover, I had to tweak `csi-node-driver-registrar` command line to make it use the correct directories, see #407

With the above changes, block volumes are working correctly on my cluster.

**Comment by rkrzewski** [comment]
Fixed with #410
